libraries/AP_NavEKF3/AP_NavEKF3_Outputs.cpp

```diff
@@ -230,7 +230,8 @@ bool NavEKF3_core::getPosNE(Vector2f &posNE) const
         // In constant position mode the EKF position states are at the origin, so we cannot use them as a position estimate
         if(validOrigin) {
             auto &gps = dal.gps();
-            if ((gps.status(selected_gps) >= AP_DAL_GPS::GPS_OK_FIX_2D)) {
+            if (frontend->sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS
```

Comment:
One concern is that the position may jump as the user changes between source sets.  So a source set that includes GPS will show the latest GPS position but if the user then switches back to external nav, it'll jump back to the last good position estimate.

I actually agree with this change or something like it.  I've always thought it odd that the EKF3 sometimes reports the raw GPS position instead of the EKF position.  We already report the raw GPS position to the GCS so I'm not sure we really need to report it again through the EKF.  That's probably too big a change for the GCSs to absorb though.

Reply:
> I actually agree with this change or something like it. I've always thought it odd that the EKF3 sometimes reports the raw GPS position instead of the EKF position. We already report the raw GPS position to the GCS so I'm not sure we really need to report it again through the EKF. That's probably too big a change for the GCSs to absorb though.

Resolving that problem is complicated by the fact that the EKF can return a position estimate even when it is feeling ill.  It will do this if it has ever managed to get an origin (i.e. be happy with where it is), and when the GPS is not capable of providing a location.

What that means is that never returning a GPS from the EKF would mean changing the interface we use onto the backends to be more complicated.   e.g. "EK3 give me a location".  What, you can't do that?  OK, "GPS, what about you?"  Really?!  OK, "EK3, please give me your best guess".  I don't think it will be all that pretty.

Reply:
... you can't really use the data if getLLH returns false; if it has never seen an origin and the GPS isn't giving a location then `false` really means that it hasn't given you anything at all.  This is the sort of case you might end up flying to 0,0